### PR TITLE
Test images distributor: Create rbac for ci-operator

### PR DESCRIPTION
**Pull request by alvaroaleman**
/hold
~needs extended rbac for the controller~ rbac was merged

### CODE REVIEW

File: pkg/controller/test-images-distributor/test_images_distributor.go [hunk 1]
```diff
@@ -323,6 +333,100 @@ func (r *reconciler) ensureImagePullSecret(namespace string, client ctrlruntimec
...
+		if err := client.Create(r.ctx, reference); err != nil {
+			return fmt.Errorf("failed to create role %s: %w", name.String(), err)
+		}
+		log.Infof("Successfully created %s role", reference.Name)
+		return nil
+	}
+	// Reset ObjectMeta to make comparable
+	rv := existingRole.ResourceVersion
+	existingRole.ObjectMeta = reference.ObjectMeta
+	if reflect.DeepEqual(existingRole, reference) {
```
Comment (stevekuznetsov) [inline pkg/controller/test-images-distributor/test_images_distributor.go:371]: Shouldn't this be `semantic.Equality.DeepEqual()`?

Comment (alvaroaleman) [reply pkg/controller/test-images-distributor/test_images_distributor.go]: Yeah, this was a bit ad hoc. I've sanitized it to use common upsert code everywhere. PTAL.

**Pull request merged**